module/data/item/templates/activities.mjs

```diff
@@ -167,46 +167,60 @@ export default class ActivitiesTemplate extends SystemDataModel {
    */
   static #migrateUses(source) {
     // Remove any old ternary operators from uses to prevent errors
-    if ( source.uses?.max?.includes(" ? ") ) source.uses.max = "";
+    let maxIncludesIsAFunction = source?.uses && typeof (source?.uses?.max?.includes) === "function";
+
     for ( const activity of Object.values(source.activities ?? {}) ) {
-      if ( activity?.uses?.max?.includes(" ? ") ) activity.uses.max = "";
+      if (maxIncludesIsAFunction && activity?.uses?.max?.includes(" ? ") )
+      {
+        activity.uses.max = "";
+      }
     }
 
-    if ( Array.isArray(source.uses?.recovery) ) return;
+    if ( Array.isArray(source?.uses?.recovery) )
+    {
+      return;
+    }
 
     const charged = source.recharge?.charged;
-    if ( (source.recharge?.value !== null) && (charged !== undefined) && !source.uses?.max ) {
+
+    if (  (source.recharge?.value !== null) && (charged !== undefined) && !source.uses?.max ) {
       source.uses ??= {};
       source.uses.spent = charged ? 0 : 1;
       source.uses.max = "1";
     }
 
-    if ( foundry.utils.getType(source.uses?.recovery) !== "string" ) return;
+    if (  foundry.utils.getType(source?.uses?.recovery) !== "string" )
+    {
+      return;
+    }
 
-    // If period is charges, set the recovery type to `formula`
-    if ( source.uses.per === "charges" ) {
+    // If period is charges, set the recovery type to formula
+    if (source?.uses?.per === "charges" ) {
```

Comment:
```suggestion
    // If period is charges, set the recovery type to `formula`
    if ( source.uses?.per === "charges" ) {
```

Reply:
On this one the optional chaining can be ditched altogether - by this point in the code we know `sources.uses.recovery` exists, which means `sources.uses` necessarily exists